ZAPD/Globals.cpp

```diff
@@ -72,27 +74,81 @@ void Globals::ReadConfigFile(const std::string& configFilePath)
 	XMLError eResult = doc.LoadFile(configFilePath.c_str());
 
 	if (eResult != tinyxml2::XML_SUCCESS)
+	{
+		throw std::runtime_error("Error: Unable to read config file.");
 		return;
+	}
 
 	XMLNode* root = doc.FirstChild();
 
 	if (root == nullptr)
 		return;
 
-	for (XMLElement* child = root->FirstChildElement(); child != NULL;
-	     child = child->NextSiblingElement())
+	cfg = new GameConfig();
+
+	for (XMLElement* child = root->FirstChildElement(); child != NULL; child = child->NextSiblingElement())
 	{
 		if (string(child->Name()) == "SymbolMap")
 		{
 			string fileName = string(child->Attribute("File"));
-			GenSymbolMap(fileName);
+			GenSymbolMap(Path::GetDirectoryName(configFilePath) + "/" + fileName);
```

Comment:
You could use a `fs::path` object to add the path separators. It has the advantage of not having to worry about double `/`.
```suggestion
			GenSymbolMap(fs::path(Path::GetDirectoryName(configFilePath)) / fileName);
```